data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/S3OutputStream.java

```diff
@@ -42,6 +46,8 @@ public class S3OutputStream extends PositionOutputStream {
      */
     private final String key;
 
+    private String targetBucket;
```

Comment:
I don't think we need this field. Just use the existing `bucket` field and make it non-final.